akka-stream/src/main/scala/akka/stream/impl/io/InputStreamPublisher.scala

```diff
@@ -87,8 +87,6 @@ private[akka] class InputStreamPublisher(is: InputStream, bytesReadPromise: Prom
 
   /** BLOCKING I/O READ */
   def loadChunk() = try {
-    val arr = Array.ofDim[Byte](chunkSize)
```

Comment:
Also, the following line: `log.debug("No more bytes available to read (got `-1` or `0` from `read`), marking final bytes of file @ " + eofReachedAtOffset)` is not true since the match only checks for `-1`.

A read of `0` should and could most definitely be optimized by the match-case.